internal/container/selector.go

```diff
@@ -12,7 +12,7 @@ const (
 )
 
 type RefSelector struct {
-	ref       reference.Named
+	Ref       reference.Named
```

Comment:
i'm confident that this shouldn't be a public field. it's not something that an API user would want or be able to use correctly. as a proof of point, the only place you did use this field is wrong and a bug.

Reply:
Not a bug, it was desired behavior -- the idea was to pass in the ref _as specified in Tiltfile_ (rather than as stripped down for deployment) so we knew whether user gave it a tag or not. But I'm redoing this PR on advice from DB, so it's moot.

Reply:
hmmm...can you point me to the spec that describes the expected behavior and how it's supposed to interact with substitute_repo?

Reply:
(my understanding from previous conversations is we expected custom_build to push to a remote repo in some cases)

Reply:
that behavior has been removed